.github/scripts/jira/enforce-jira-issue.ts

```diff
@@ -0,0 +1,77 @@
+import * as core from "@actions/core";
+import jira from "jira.js";
+import { createJiraClient, parseIssueNumberFrom } from "./lib";
+
+async function doesIssueExist(
+  client: jira.Version3Client,
+  issueNumber: string,
+  dryRun: boolean
+) {
+  const payload = {
+    issueIdOrKey: issueNumber,
+  };
+
+  if (dryRun) {
+    core.info("Dry run enabled, skipping JIRA issue enforcement");
+    return true;
+  }
+
+  try {
+    /**
+     * The issue is identified by its ID or key, however, if the identifier doesn't match an issue, a case-insensitive search and check for moved issues is performed.
+     * If a matching issue is found its details are returned, a 302 or other redirect is not returned. The issue key returned in the response is the key of the issue found.
+     */
+    const issue = await client.issues.getIssue(payload);
+    core.debug(
+      `JIRA issue id:${issue.id} key: ${issue.key} found while querying for ${issueNumber}`
+    );
+    if (issue.key !== issueNumber) {
+      core.error(
+        `JIRA issue key ${issueNumber} not found, but found issue key ${issue.key} instead`
```

Comment:
maybe in this case let's ask the user to either contact with JIRA administrator or create a new ticket, because this should never happen?

Reply:
it could be someone typed a typo for the ticket id?